framework/src/modules/chain/logic/dapp.js

```diff
@@ -37,19 +36,24 @@ __private.unconfirmedAscii = {};
  * @requires bytebuffer
  * @requires valid
  * @requires helpers/dapp_categories
- * @param {Storage} storage
- * @param {Object} logger
- * @param {ZSchema} schema
- * @param {Object} network
+ * @param {Object} dependencies
+ * @param {Object} dependencies.components
+ * @param {Object} dependencies.libraries
+ * @param {Storage} dependencies.components.storage
+ * @param {logger} dependencies.components.logger
+ * @param {ZSchema} dependencies.libraries.schema
+ * @param {Object} dependencies,libraries.network
  * @todo Add description for the params
  */
 class DApp {
-	constructor(storage, logger, schema, network) {
-		library = {
-			storage,
-			logger,
-			schema,
-			network,
+	constructor({ components, libraries }) {
```

Comment:
Can we do it more explicit with deep object decomposition. 

```
constructor({ components: { logger, storage }, network, schema }) {
```

Reply:
I like it this way, and testing will be easier too. @diego-G this solves the default values problem.